sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamEnumerableConverter.java

```diff
@@ -304,7 +305,11 @@ private static Object fieldToAvatica(Schema.FieldType type, Object beamValue) {
                   MetricsFilter.builder()
                       .addNameFilter(MetricNameFilter.named(BeamEnumerableConverter.class, "rows"))
                       .build());
-      count = metrics.getCounters().iterator().next().getAttempted();
+      try {
+        count = metrics.getCounters().iterator().next().getAttempted();
+      } catch (NoSuchElementException e) {
```

Comment:
i think it's better to guard `.next()` with `.hasNext()` instead of catching the exception

Reply:
Updated.

Thought about it. Your approach is better from the perspective of avoiding catching other NoSuchElementExceptions that are not caused by this issue.